client.go

```diff
@@ -309,6 +309,9 @@ func (c *Client) makeStreamError(statusCode int, body io.Reader) error {
 	if err := json.Unmarshal(errBody, &streamErr); err != nil {
 		return fmt.Errorf("unexpected error (status code %d)", statusCode)
 	}
+	if streamErr.Detail == "" {
```

Comment:
We won't reach this branch because we return an error on JSON parsing failure
Example: https://play.golang.org/p/9mRLXGtS6OA

Instead we need to try to handle this in the error branch